### PR TITLE
Fix #223: Refactor OnboardingProvider to use context objects

### LINKED ISSUE #223: Refactor OnboardingProvider to use context objects
Refactor OnboardingProvider to use context objects.

### PR DESCRIPTION
**Pull request by banterCZ**

### CODE REVIEW

File: enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/provider/OnboardingProvider.java [hunk 1]
```diff
@@ -19,17 +19,15 @@
...
  *
  * @author Roman Strobl, roman.strobl@wultra.com
  */
 public interface OnboardingProvider {
 
-    String lookupUser(Map<String, Object> identification) throws OnboardingProviderException;
+    String lookupUser(LookupUserRequest request) throws OnboardingProviderException;
 
-    void sendOtpCode(String userId, String otpCode, boolean resend) throws OnboardingProviderException;
+    void sendOtpCode(SendOtpCodeRequest request) throws OnboardingProviderException;
```
Comment (romanstrobl) [inline enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/provider/OnboardingProvider.java:31]: Please add JavaDoc.

**Review by romanstrobl — APPROVED** [review]
Looks good, but I found some minor possible improvements.

File: enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/provider/LookupUserRequest.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ * Request object for {@link OnboardingProvider#lookupUser(LookupUserRequest)}.
+ *
+ * @author Lubos Racansky, lubos.racansky@wultra.com
+ */
+@Builder
+@Getter
+// TODO (racansky, 2022-06-17) @PublicApi
+public final class LookupUserRequest {
+
+    private Map<String, Object> identification;
```
[the remark this answers is not in the document]
Comment (banterCZ) [reply enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/provider/LookupUserRequest.java:35]: Good point but final or immutable? Builder may set it null but I will try to add validation annotation.

[the remark this answers is not in the document]
Comment (banterCZ) [reply enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/provider/LookupUserRequest.java:37]: Fixed by `@lombok.NotNull`.

**Pull request merged**